### ISSUE TITLE
add ToC in doc pages

### ISSUE THREAD

**Issue by tnielens** [pityka/saddle#319]
The current hugo theme does not support page tocs. Switching to a doc theme that supports it would improve the doc navigability. 
Example of page toc here: https://doks.netlify.app/docs/prologue/introduction/

Two potential alternatives:
- switch to an hugo template that supports it. This [one](https://doks.netlify.app/docs/prologue/introduction/) or this [one](https://hugo-book-demo.netlify.app/) are great candidates.
- switch to [docausaurus](https://docusaurus.io/) and leverage [mdoc integrations](https://scalameta.org/mdoc/docs/docusaurus.html).

Cherry on the cake: all these doc templates have search bars.

My suggestion is a switch to docausaurus and mdoc integration because many other reference libraries in the scala ecosystem use it.

I'd be glad to take this task.

**Comment by pityka** [comment]
I would like to keep hugo, so if you can replace the theme with one that you think it suits better I would be thankful. The doks theme looks nice indeed.

**Comment by tnielens** [comment]
I've tried the doks theme for a while. The site look was great but it was more complex to setup than the whisper or book themes. It deviates from the standard hugo theme mechanism and add a dependency on `npm`. The branch I experimented with is [here](https://github.com/tnielens/saddle/tree/hugo-doks-theme). I would suggest not to use it due to its added complexity in comparison with hugo-whisper or hugo-book given saddle's simple doc site needs. 

On the other hand, hugo-good was very easy to setup and offers nice improvements over whisper. It's in pr for review.
